playbooks/preflight-check.yml

```diff
@@ -0,0 +1,64 @@
+---
+# Copyright 2017, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Preflight check
+  hosts: "all_containers"
+  gather_facts: false
+  user: root
+  tasks:
+    - name: Ping check
+      ping:
+      register: ping_result
+      failed_when: "'FAILED' in ping_result"
+    - debug:
+        msg: "{{ ping_result.ping }}"
+
+- name: Openstack release check
+  hosts: "hosts"
+  gather_facts: false
+  user: root
+  tasks:
+    - name: Release should be lower than Newton
+      shell: grep 'DISTRIB_RELEASE' /etc/openstack-release | sed -e 's/.*"\(.*\)"/\1/'
+      register: openstack_release_return
+      failed_when: "{{ openstack_release_return.stdout | version_compare('14.0.0', '>=') }}"
+
+- name: Ceph check
+  hosts: "hosts"
+  gather_facts: false
+  user: root
+  vars:
+    file_content: "{{ lookup('file', '/opt/rpc-openstack/openstack-ansible/scripts/inventory-manage.py') }}"
+  tasks:
+    - name: check ceph exists
+      fail:
+        msg: "Ceph group found, the leapfrog can't continue."
```

Comment:
We might want to say something like *NotImplemented: leap upgrades where ceph is deployed is not implemented at this time*.

Reply:
Sure, got it. It's more official and accurate.